marshaller_test.go

```diff
@@ -1096,6 +1107,22 @@ var _ = Describe("Marshaller", func() {
 						Expect(str).To(Equal("Id,Name__c,NonNestedStruct__r.Name,NonNestedStruct__r.SomeValue__c"))
 					})
 				})
+
+				Context("when nested struct with soql fields is selected opaquely", func() {
```

Comment:
Can you add a test that verifies the entire SOQL generation? Basically add a `Context` to `Describe("Marshal")`

Reply:
I will look into it 👍